compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeWindowLayer.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.awt
+
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.Graphics
+import javax.accessibility.Accessible
+import javax.swing.SwingUtilities
+import org.jetbrains.skiko.ClipRectangle
+import org.jetbrains.skiko.GraphicsApi
+import org.jetbrains.skiko.SkiaLayer
+import org.jetbrains.skiko.SkiaLayerAnalytics
+
+/**
+ * Provides a heavyweight AWT [component] used to render content (from [setContent]) on-screen with Skia.
+ *
+ * If smooth interop with Swing is needed, consider using [androidx.compose.ui.awt.ComposeWindowLayer]
+ */
+internal class ComposeWindowLayer(
+    private val skiaLayerAnalytics: SkiaLayerAnalytics
+) : ComposeLayer<SkiaLayer>() {
+    private val _component = ComposeWindowSkiaLayer()
+    override val component: SkiaLayer get() = _component
+
+    override val renderApi: GraphicsApi
+        get() = _component.renderApi
+
+    override val clipComponents: MutableList<ClipRectangle>
+        get() = _component.clipComponents
+
+    override val focusComponentDelegate: Component
+        get() = _component.canvas
+
+    init {
+        _component.skikoView = skikoView
+        attachComposeToComponent()
+    }
+
+    override fun requestNativeFocusOnAccessible(accessible: Accessible) {
+        _component.requestNativeFocusOnAccessible(accessible)
+    }
+
+    override fun onComposeInvalidation() {
+        _component.needRedraw()
+    }
+
+    override fun disposeComponentLayer() {
+        _component.dispose()
+    }
+
+    /**
+     * See also backend layer for swing interop: [androidx.compose.ui.awt.ComposeSwingLayer.ComposeSwingSkiaLayer]
+     */
+    private inner class ComposeWindowSkiaLayer :
+        SkiaLayer(
+            externalAccessibleFactory = { sceneAccessible },
+            analytics = skiaLayerAnalytics
+        ),
+        Accessible {
+        override fun addNotify() {
```

Comment:
it took some time to understand why it formatted this way 😅

Reply:
Could you please ellaborate? How would you prefer?

Reply:
See [Coding conventions](https://kotlinlang.org/docs/coding-conventions.html#class-headers):

For classes with a long supertype list, put a line break after the colon and align all supertype names horizontally:
```
class MyFavouriteVeryLongClassHolder :
    MyLongHolder<MyFavouriteVeryLongClass>(),
    SomeOtherInterface,
    AndAnotherOne {

    fun foo() { /*...*/ }
}
```

To clearly separate the class header and body when the class header is long, either put a blank line following the class header (as in the example above), or put the opening curly brace on a separate line:
```
class MyFavouriteVeryLongClassHolder :
    MyLongHolder<MyFavouriteVeryLongClass>(),
    SomeOtherInterface,
    AndAnotherOne
{
    fun foo() { /*...*/ }
}
```